File: backend/HRManagementSystem/src/Core/HRManagementSystem.Domain/Entities/Allowance.cs
namespace HRManagementSystem.Domain.Entities;

public class Allowance:BaseEntity
{
    public string Name { get; set; } = default!;
    public decimal Amount { get; set; }
    public bool IsPercentage { get; set; } = false;
    public ICollection<EmployeeAllowance> EmployeeAllowances { get; set; } = new List<EmployeeAllowance>();


}

File: frontend/src/app/models/allowance.ts
export interface AllowanceDto {
    id: number;
    name: string;
    amount: number;
    isPercentage: boolean;
}

export interface CreateAllowanceDto {
    name: string;
    amount: number;
    isPercentage: boolean;
}

export interface UpdateAllowanceDto {
    name: string;
    amount: number;
    isPercentage: boolean;
}
